### PR TITLE
support encrypted/compressed objects properly during decommission

### LINKED ISSUE #15314: Decomissioning fails, possibly due to IncompleteBody errors
I'm currently trying to decommission a pool in a MinIO cluster with two pools. The decommissioning always fails.

Before decommissioning, the pool had over 500 GB used per drive. Now it has 228 GB used per drive.

When trying to decommission again, MinIO seems to iterate over all the objects, but doesn't remove any more objects from the pool, because it stays at 228 GB.

During decommissioning (both initially and now when running it again), there are a lot of these errors in the MinIO log. (Speculation: perhaps this error occurs with every object in the pool and that's why MinIO doesn't remove them?)

```
minio-old-zone0-0 minio API: SYSTEM()
minio-old-zone0-0 minio Time: 11:51:07 UTC 07/16/2022
minio-old-zone0-0 minio DeploymentID: 73a34713-7b8c-4d45-98ba-6e9c6a8cc1df
minio-old-zone0-0 minio Error: cluster01-gitlab-registry/docker/registry/v2/blobs/sha256/d7/d7b6e6950809eba24f1016ab5db1f80078bcf58caf57ac47c626630e462eae86/data has incomplete body (cmd.IncompleteBody)
minio-old-zone0-0 minio        2: internal/logger/logger.go:270:logger.LogIf()
minio-old-zone0-0 minio        1: cmd/erasure-server-pool-decom.go:745:cmd.(*erasureServerPools).decommissionPool.func1()
minio-old-zone0-0 minio 
```

All of the objects where this error occurs seem to be readable (e.g. with `mc cat`) just fine. So they are not corrupted.

## Expected Behavior

Decommissioning should succeed to migrate those objects.

## Your Environment

MinIO 2022-07-13T23:29:44Z on Kubernetes via containerd, bare metal, linux `5.10.0-15-amd64 #1 SMP Debian 5.10.120-1 (2022-06-09) x86_64 GNU/Linux`

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
support encrypted/compressed objects properly during decommission

## Motivation and Context
decommission did not support encrypted and compressed objects
properly due to incorrect length calculation.

fixes #15314 
## How to test this PR?
Unit tests have been added to test the relevant functionality.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [x] Unit tests added/updated

### CODE REVIEW

File: cmd/object-api-utils.go [hunk 1]
```diff
@@ -646,8 +646,9 @@ func NewGetObjectReader(rs *HTTPRangeSpec, oi ObjectInfo, opts ObjectOptions) (
 		return nil, 0, 0, err
 	}
 
-	// if object is encrypted and it is a restore request, fetch content without decrypting.
-	if opts.Transition.RestoreRequest != nil {
+	// if object is encrypted and it is a restore request, fetch content
+	// without decrypting or if NoDecryption was requested.
```
Comment (haslersn) [inline cmd/object-api-utils.go]:
```suggestion
	// if object is encrypted and it is a restore request or if NoDecryption was requested,
	// fetch content without decrypting.
```

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#043c4b08-a77d-45b7-a1a5-4142a00b4bea) |
| mint-fs.sh | :x: [more...](#652bd915-3cab-4480-91b7-465ccf85fff0) |
| mint-erasure.sh | :x: [more...](#ca9b75b0-5fc3-4add-901d-8c968b602f5d) |
| mint-dist-erasure.sh | :x: [more...](#49a8ab46-5039-4f0d-a8df-b5ca243fd267) |
| mint-compress-encrypt-dist-erasure.sh | :x: [more...](#d7a6c7eb-c861-416f-b015-9a3ea2ad69c0) |
<a name="49a8ab46-5039-4f0d-a8df-b5ca243fd267"></a>
### 15320-b3dd4b7/mint-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s7:31629
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/14) Running awscli tests ... done in 1 minutes and 1 seconds
(2/14) Running aws-sdk-go tests ... done in 1 seconds
(3/14) Running aws-sdk-java tests ... done in 1 seconds
(4/14) Running aws-sdk-php tests ... done in 44 seconds
(5/14) Running aws-sdk-ruby tests ... done in 6 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 28 seconds
(8/14) Running minio-go tests ... done in 1 minutes and 7 seconds
(9/14) Running minio-java tests ... done in 30 seconds
(10/14) Running minio-js tests ... FAILED in 1 minutes and 25 seconds
{
  "name": "minio-js",
  "function": "listenBucketNotification(bucketName, prefix, suffix, events)",
  "args": "bucketName:minio-js-test-e23263b9-22d9-4275-b4af-f86fda89a2f4, events: s3:ObjectCreated:*",
  "duration": 65106,
  "status": "FAIL",
  "error": "Error: socket hang up at connResetException (internal/errors.js:639:14) at Socket.socketOnEnd (_http_client.js:499:23) at endReadableNT (internal/streams/readable.js:1333:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)"
}
(10/14) Running minio-py tests ... done in 1 minutes and 14 seconds
(11/14) Running s3cmd tests ... done in 25 seconds
(12/14) Running s3select tests ... done in 3 seconds
(13/14) Running versioning tests ... done in 3 minutes and 4 seconds

Executed 13 out of 14 tests successfully.
```
<a name="652bd915-3cab-4480-91b7-465ccf85fff0"></a>
### 15320-b3dd4b7/mint-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s1:32411
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/14) Running awscli tests ... done in 52 seconds
(2/14) Running aws-sdk-go tests ... done in 0 seconds
(3/14) Running aws-sdk-java tests ... done in 1 seconds
(4/14) Running aws-sdk-php tests ... done in 43 seconds
(5/14) Running aws-sdk-ruby tests ... done in 5 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 21 seconds
(8/14) Running minio-go tests ... done in 1 minutes and 14 seconds
(9/14) Running minio-java tests ... FAILED in 5 minutes and 17 seconds
(9/14) Running minio-js tests ... done in 59 seconds
(10/14) Running minio-py tests ... done in 46 seconds
(11/14) Running s3cmd tests ... done in 22 seconds
(12/14) Running s3select tests ... done in 2 seconds
(13/14) Running versioning tests ... done in 3 minutes and 1 seconds

Executed 13 out of 14 tests successfully.
```
<a name="d7a6c7eb-c861-416f-b015-9a3ea2ad69c0"></a>
### 15320-b3dd4b7/mint-compress-encrypt-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s5:30752
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/14) Running awscli tests ... done in 1 minutes and 6 seconds
(2/14) Running aws-sdk-go tests ... done in 1 seconds
(3/14) Running aws-sdk-java tests ... done in 1 seconds
(4/14) Running aws-sdk-php tests ... done in 44 seconds
(5/14) Running aws-sdk-ruby tests ... done in 7 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 29 seconds
(8/14) Running minio-go tests ... done in 1 minutes and 27 seconds
(9/14) Running minio-java tests ... FAILED in 1 minutes and 56 seconds
{
  "name": "minio-java",
  "function": "listObjects()",
  "args": "[bucket]",
  "duration": 110607,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = NoSuchBucket, message = The specified bucket does not exist, bucketName = minio-java-test-3jvjitr, objectName = null, resource = /minio-java-test-3jvjitr, requestId = 17026C02028B1893, hostId = 3887a1c7-ef26-4f13-9a78-a86afbdee404)\nrequest={method=DELETE, url=http://minio-k8s5:30752/minio-java-test-3jvjitr, headers=Host: minio-k8s5:30752\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.4.3\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20220716T211738Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20220716/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=404, headers=Accept-Ranges: bytes\nContent-Length: 318\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO\nStrict-Transport-Security: max-age=31536000; includeSubDomains\nVary: Origin\nVary: Accept-Encoding\nX-Amz-Request-Id: 17026C02028B1893\nX-Content-Type-Options: nosniff\nX-Xss-Protection: 1; mode=block\nDate: Sat, 16 Jul 2022 21:19:28 GMT\n}\n >>> [io.minio.S3Base$1.onResponse(S3Base.java:690), okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519), java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128), java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628), java.base/java.lang.Thread.run(Thread.java:829)]"
}
(9/14) Running minio-js tests ... done in 1 minutes and 5 seconds
(10/14) Running minio-py tests ... done in 1 minutes and 18 seconds
(11/14) Running s3cmd tests ... done in 35 seconds
(12/14) Running s3select tests ... done in 3 seconds
(13/14) Running versioning tests ... done in 3 minutes and 5 seconds

Executed 13 out of 14 tests successfully.
```
<a name="ca9b75b0-5fc3-4add-901d-8c968b602f5d"></a>
### 15320-b3dd4b7/mint-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s2:30981
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/14) Running awscli tests ... done in 56 seconds
(2/14) Running aws-sdk-go tests ... done in 0 seconds
(3/14) Running aws-sdk-java tests ... done in 1 seconds
(4/14) Running aws-sdk-php tests ... done in 43 seconds
(5/14) Running aws-sdk-ruby tests ... done in 5 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 25 seconds
(8/14) Running minio-go tests ... done in 1 minutes and 18 seconds
(9/14) Running minio-java tests ... done in 23 seconds
(10/14) Running minio-js tests ... FAILED in 1 minutes and 26 seconds
{
  "name": "minio-js",
  "function": "listenBucketNotification(bucketName, prefix, suffix, events)",
  "args": "bucketName:minio-js-test-594bc3b3-3da1-496c-83a3-0d330d6992dd, events: s3:ObjectCreated:*",
  "duration": 65691,
  "status": "FAIL",
  "error": "Error: socket hang up at connResetException (internal/errors.js:639:14) at Socket.socketOnEnd (_http_client.js:499:23) at endReadableNT (internal/streams/readable.js:1333:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)"
}
(10/14) Running minio-py tests ... done in 52 seconds
(11/14) Running s3cmd tests ... done in 24 seconds
(12/14) Running s3select tests ... done in 2 seconds
(13/14) Running versioning tests ... done in 3 minutes and 2 seconds

Executed 13 out of 14 tests successfully.
```
<a name="043c4b08-a77d-45b7-a1a5-4142a00b4bea"></a>
### 15320-b3dd4b7/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s1:30742
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/14) Running awscli tests ... done in 1 minutes and 1 seconds
(2/14) Running aws-sdk-go tests ... done in 1 seconds
(3/14) Running aws-sdk-java tests ... done in 1 seconds
(4/14) Running aws-sdk-php tests ... done in 44 seconds
(5/14) Running aws-sdk-ruby tests ... done in 6 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 29 seconds
(8/14) Running minio-go tests ... done in 1 minutes and 31 seconds
(9/14) Running minio-java tests ... FAILED in 2 minutes and 27 seconds
{
  "name": "minio-java",
  "function": "removeBucket()",
  "duration": 5,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = BucketNotEmpty, message = The bucket you tried to delete is not empty, bucketName = minio-java-test-1if203l, objectName = null, resource = /minio-java-test-1if203l, requestId = 17026C03F4FDBF2A, hostId = 5ce48230-65cc-4eaf-8adc-24579ec285eb)\nrequest={method=DELETE, url=http://minio-k8s1:30742/minio-java-test-1if203l, headers=Host: minio-k8s1:30742\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.4.3\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20220716T211937Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20220716/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=409, headers=Accept-Ranges: bytes\nContent-Length: 328\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO\nStrict-Transport-Security: max-age=31536000; includeSubDomains\nVary: Origin\nVary: Accept-Encoding\nX-Amz-Request-Id: 17026C03F4FDBF2A\nX-Content-Type-Options: nosniff\nX-Xss-Protection: 1; mode=block\nDate: Sat, 16 Jul 2022 21:19:37 GMT\n}\n >>> [io.minio.S3Base$1.onResponse(S3Base.java:690), okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519), java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128), java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628), java.base/java.lang.Thread.run(Thread.java:829)]"
}
(9/14) Running minio-js tests ... done in 1 minutes and 4 seconds
(10/14) Running minio-py tests ... done in 1 minutes and 23 seconds
(11/14) Running s3cmd tests ... done in 25 seconds
(12/14) Running s3select tests ... done in 3 seconds
(13/14) Running versioning tests ... done in 3 minutes and 5 seconds

Executed 13 out of 14 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :x: [more...](#273ff932-a142-4d14-a92f-1a274c17b033) |
<a name="273ff932-a142-4d14-a92f-1a274c17b033"></a>
### 15320-998ea5d/mint-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s3:32326
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/14) Running awscli tests ... done in 1 minutes and 0 seconds
(2/14) Running aws-sdk-go tests ... done in 1 seconds
(3/14) Running aws-sdk-java tests ... done in 2 seconds
(4/14) Running aws-sdk-php tests ... done in 43 seconds
(5/14) Running aws-sdk-ruby tests ... done in 6 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 38 seconds
(8/14) Running minio-go tests ... FAILED in 44 seconds
{
  "args": {
    "bucketName": "minio-go-test-566asobye9ot5h4s",
    "expires": 3600000000000,
    "objectName": "minio-go-test-566asobye9ot5h4sunique",
    "reqParams": {
      "response-content-disposition": [
        "attachment; filename=\"test.txt\""
      ]
    }
  },
  "duration": 35236,
  "error": "Get \"http://minio-k8s3:32326/minio-go-test-566asobye9ot5h4s/minio-go-test-566asobye9ot5h4sunique?AWSAccessKeyId=minio&Expires=1658011062&response-content-disposition=attachment%3B%20filename%3D%22test.txt%22&Signature=2whTWPbNmhlhjDy1/C613%2B2f4MA%3D\": context deadline exceeded (Client.Timeout exceeded while awaiting headers)",
  "function": "PresignedGetObject(bucketName, objectName, expires, reqParams)",
  "message": "PresignedGetObject response incorrect",
  "name": "minio-go: testFunctionalV2",
  "status": "FAIL"
}
(8/14) Running minio-java tests ... done in 54 seconds
(9/14) Running minio-js tests ... done in 1 minutes and 4 seconds
(10/14) Running minio-py tests ... done in 1 minutes and 20 seconds
(11/14) Running s3cmd tests ... done in 25 seconds
(12/14) Running s3select tests ... done in 4 seconds
(13/14) Running versioning tests ... done in 3 minutes and 4 seconds

Executed 13 out of 14 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-fs.sh | :x: [more...](#9aa73b5e-fcbb-4046-a3bc-4d16fb01cbff) |
| mint-erasure.sh | :x: [more...](#c88f83c2-f0c9-4b9d-9237-2413d12a648a) |
| mint-pools.sh | :x: [more...](#f5412c30-5074-4a2c-84da-bd068f768178) |
<a name="9aa73b5e-fcbb-4046-a3bc-4d16fb01cbff"></a>
### 15320-39814b5/mint-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s5:31674
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/14) Running awscli tests ... done in 1 minutes and 1 seconds
(2/14) Running aws-sdk-go tests ... done in 0 seconds
(3/14) Running aws-sdk-java tests ... done in 1 seconds
(4/14) Running aws-sdk-php tests ... done in 43 seconds
(5/14) Running aws-sdk-ruby tests ... done in 5 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 22 seconds
(8/14) Running minio-go tests ... done in 1 minutes and 7 seconds
(9/14) Running minio-java tests ... done in 24 seconds
(10/14) Running minio-js tests ... FAILED in 2 minutes and 25 seconds
{
  "name": "minio-js",
  "function": "listenBucketNotification(bucketName, prefix, suffix, events)",
  "args": "bucketName:minio-js-test-fcefdc0c-7687-46d3-9497-78a8680997e6, events: s3:ObjectCreated:*",
  "duration": 126675,
  "status": "FAIL",
  "error": "Error: read ECONNRESET at TCP.onStreamRead (internal/stream_base_commons.js:209:20)"
}
(10/14) Running minio-py tests ... done in 47 seconds
(11/14) Running s3cmd tests ... done in 22 seconds
(12/14) Running s3select tests ... done in 2 seconds
(13/14) Running versioning tests ... done in 3 minutes and 1 seconds

Executed 13 out of 14 tests successfully.
```
<a name="f5412c30-5074-4a2c-84da-bd068f768178"></a>
### 15320-39814b5/mint-pools.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s1:32531
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/14) Running awscli tests ... done in 56 seconds
(2/14) Running aws-sdk-go tests ... done in 1 seconds
(3/14) Running aws-sdk-java tests ... done in 1 seconds
(4/14) Running aws-sdk-php tests ... done in 45 seconds
(5/14) Running aws-sdk-ruby tests ... done in 7 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 33 seconds
(8/14) Running minio-go tests ... done in 1 minutes and 32 seconds
(9/14) Running minio-java tests ... FAILED in 1 minutes and 14 seconds
{
  "name": "minio-java",
  "function": "listObjects()",
  "args": "[bucket, prefix, recursive, 1050 objects]",
  "duration": 66109,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = BucketNotEmpty, message = The bucket you tried to delete is not empty, bucketName = minio-java-test-2cq7slu, objectName = null, resource = /minio-java-test-2cq7slu, requestId = 17026E23A7F8AFBD, hostId = 2e600cf6-9ff8-4ff5-ba0f-c88b8becd375)\nrequest={method=DELETE, url=http://minio-k8s1:32531/minio-java-test-2cq7slu, headers=Host: minio-k8s1:32531\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.4.3\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20220716T215832Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20220716/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=409, headers=Accept-Ranges: bytes\nContent-Length: 328\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO\nStrict-Transport-Security: max-age=31536000; includeSubDomains\nVary: Origin\nVary: Accept-Encoding\nX-Amz-Request-Id: 17026E23A7F8AFBD\nX-Content-Type-Options: nosniff\nX-Xss-Protection: 1; mode=block\nDate: Sat, 16 Jul 2022 21:58:32 GMT\n}\n >>> [io.minio.S3Base$1.onResponse(S3Base.java:690), okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519), java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128), java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628), java.base/java.lang.Thread.run(Thread.java:829)]"
}
(9/14) Running minio-js tests ... done in 1 minutes and 6 seconds
(10/14) Running minio-py tests ... done in 1 minutes and 34 seconds
(11/14) Running s3cmd tests ... done in 26 seconds
(12/14) Running s3select tests ... done in 4 seconds
(13/14) Running versioning tests ... done in 3 minutes and 5 seconds

Executed 13 out of 14 tests successfully.
```
<a name="c88f83c2-f0c9-4b9d-9237-2413d12a648a"></a>
### 15320-39814b5/mint-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s2:31990
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/14) Running awscli tests ... done in 53 seconds
(2/14) Running aws-sdk-go tests ... done in 1 seconds
(3/14) Running aws-sdk-java tests ... done in 1 seconds
(4/14) Running aws-sdk-php tests ... done in 42 seconds
(5/14) Running aws-sdk-ruby tests ... done in 6 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 21 seconds
(8/14) Running minio-go tests ... done in 58 seconds
(9/14) Running minio-java tests ... done in 29 seconds
(10/14) Running minio-js tests ... FAILED in 2 minutes and 43 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 2,
  "status": "FAIL",
  "error": "Error: read ECONNRESET at TCP.onStreamRead (internal/stream_base_commons.js:209:20)"
}
(10/14) Running minio-py tests ... done in 50 seconds
(11/14) Running s3cmd tests ... done in 23 seconds
(12/14) Running s3select tests ... done in 2 seconds
(13/14) Running versioning tests ... done in 3 minutes and 1 seconds

Executed 13 out of 14 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
| mint-erasure.sh | :x: [more...](#04a56f80-ab48-47e9-ae90-7b3a143625fb) |
<a name="04a56f80-ab48-47e9-ae90-7b3a143625fb"></a>
### 15320-39814b5/mint-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s6:31970
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/14) Running awscli tests ... done in 58 seconds
(2/14) Running aws-sdk-go tests ... done in 1 seconds
(3/14) Running aws-sdk-java tests ... done in 1 seconds
(4/14) Running aws-sdk-php tests ... done in 42 seconds
(5/14) Running aws-sdk-ruby tests ... done in 5 seconds
(6/14) Running healthcheck tests ... done in 1 seconds
(7/14) Running mc tests ... done in 25 seconds
(8/14) Running minio-go tests ... done in 57 seconds
(9/14) Running minio-java tests ... FAILED in 1 minutes and 11 seconds
{
  "name": "minio-java",
  "function": "getObject()",
  "args": "[multi-part upload, offset, length]",
  "duration": 55791,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = NoSuchKey, message = The specified key does not exist., bucketName = minio-java-test-38avab2, objectName = minio-java-test-1ass8js, resource = /minio-java-test-38avab2/minio-java-test-1ass8js, requestId = 1702705462D1A527, hostId = 55f208a0-dae5-4725-9452-d225c0bd9a1a)\nrequest={method=GET, url=http://minio-k8s6:31970/minio-java-test-38avab2/minio-java-test-1ass8js, headers=Range: bytes=1000-66535\nHost: minio-k8s6:31970\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.4.3\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20220716T223840Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20220716/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;range;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=404, headers=Accept-Ranges: bytes\nContent-Length: 371\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nNo-Gzip-Compression: true\nServer: MinIO\nStrict-Transport-Security: max-age=31536000; includeSubDomains\nVary: Origin\nVary: Accept-Encoding\nX-Amz-Request-Id: 1702705462D1A527\nX-Content-Type-Options: nosniff\nX-Xss-Protection: 1; mode=block\nDate: Sat, 16 Jul 2022 22:38:40 GMT\n}\n >>> [io.minio.S3Base$1.onResponse(S3Base.java:690), okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519), java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128), java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628), java.base/java.lang.Thread.run(Thread.java:829)]"
}
(9/14) Running minio-js tests ... done in 1 minutes and 1 seconds
(10/14) Running minio-py tests ... done in 51 seconds
(11/14) Running s3cmd tests ... done in 24 seconds
(12/14) Running s3select tests ... done in 1 seconds
(13/14) Running versioning tests ... done in 3 minutes and 2 seconds

Executed 13 out of 14 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Review by poornas — APPROVED** [review]

**Pull request merged**